src/rvsdg/tree_unique/to_tree.rs

```diff
@@ -103,6 +114,17 @@ fn bind_value(expr: RcExpr, body: RcExpr) -> RcExpr {
 }
 
 impl<'a> RegionTranslator<'a> {
+    /// Adds a pure expression to the cache.
+    fn add_pure_value(&mut self, expr: RcExpr, id: Id) -> StoredNode {
+        if self.optimize_lets {
+            let res = StoredValue::Expr(expr.clone());
```

Comment:
It'd help to have a comment here explaining what's happening.

It sounds like we are essentially "inlining" the expression rather than binding it, is that right?

Reply:
Yes, we are inlining instead of binding. I'll add a comment